src/backend/distributed/planner/multi_logical_optimizer.c

```diff
@@ -2575,6 +2585,41 @@ ProcessWindowFunctionsForWorkerQuery(List *windowClauseList,
 }
 
 
+/* ProcessWindowFunctionPullUpForWorkerQuery pulls up inputs for window functions */
+static void
+ProcessWindowFunctionPullUpForWorkerQuery(MultiExtendedOp *originalOpNode,
+										  QueryTargetList *queryTargetList)
+{
+	if (originalOpNode->windowClause != NIL)
+	{
+		List *columnList = pull_var_clause_default((Node *) originalOpNode->windowClause);
+		StringInfoData columnNameString;
+		initStringInfo(&columnNameString);
+
+		Expr *newExpression = NULL;
+		foreach_ptr(newExpression, columnList)
+		{
+			TargetEntry *newTargetEntry = makeNode(TargetEntry);
+
+			newTargetEntry->expr = newExpression;
+
+			resetStringInfo(&columnNameString);
+			appendStringInfo(&columnNameString, WORKER_COLUMN_FORMAT,
+							 queryTargetList->targetProjectionNumber);
```

Comment:
Would be nice to capture column name generation in a function.

Reply:
This is also an incorrect use of `resetStringInfo`. Will update with 3 uses of `WORKER_COLUMN_FORMAT` using a function